### PR TITLE
MOB-94 Add invalid events filter

### PR DESCRIPTION
**Pull request by LcTwisk**
Invalid events tend to get lost when using NepSnowplow for a while. Therefore, we're adding an option to filter out valid events, which makes it easier to spot bad events that require attention.

https://user-images.githubusercontent.com/3274157/208129048-6c7f8329-90c6-42bd-bfb3-c32d0ac5b447.mov

### CODE REVIEW

**Review by FlorBosch — APPROVED** [review]

File: server/filter.js [hunk 1]
```diff
@@ -1,97 +1,82 @@
 "use strict";
 const remote = require("@electron/remote");
+const { debounce } = require("./utils");
+var filterQuery = "",
+    filterValidEvents = false;
 
-var filterTimer,
-    filterState = false;
+var update = debounce(filterItems, 50);
```
Comment (johnaziz57) [inline server/filter.js:7]: Shouldn't this be `const`?

File: server/filter.js [hunk 2]
```diff
@@ -1,97 +1,82 @@
 "use strict";
 const remote = require("@electron/remote");
+const { debounce } = require("./utils");
+var filterQuery = "",
```
Comment (johnaziz57) [inline server/filter.js:4]: Shouldn't this be `const`?

File: server/filter.js [hunk 3]
```diff
@@ -1,97 +1,82 @@
...
-function filterItems(items, value) {
-    if (typeof value === "undefined") {
-        value = document.getElementById("filter-events").value;
-    }
-    if (value === "") {
-        clearFilter();
-    }
-    // we're filtering case insensitive
-    value = value.toLowerCase();
+function filterItems() {
```
Comment (johnaziz57) [inline server/filter.js:9]: I am getting this error when looking at the code.
Should `filterItems()`  `getSchemaName()` and `contains()` be moved to the top?

Screenshot 2022-12-20 at 09 40 22

File: server/filter.js [hunk 4]
```diff
@@ -1,97 +1,82 @@
...
 
-    let events = remote.getGlobal("trackedEvents");
-    Array.from(items).forEach((eventEl) => {
-        let index = eventEl.id.substring("events-".length - 1);
-        let event = events[index];
-        let match = !filterState; // when not filtering, assume matched
+    Array.from(eventItems).forEach((eventEl) => {
+        const index = eventEl.id.substring("events-".length - 1);
+        const event = events[index];
+        var match = !filterQuery; // when not filtering, assume matched
```
Comment (johnaziz57) [inline server/filter.js:16]: we should change this to `let` 😁

**Comment by LcTwisk** [comment]
@johnaziz57 Currently, we don't use any linter on this project. Will add a ticket to add one, which will also solve all `var` vs `const`/`let` usages in one go.

**Pull request merged**